wormhole/dial.go

```diff
@@ -146,8 +149,22 @@ func (c *Wormhole) Write(p []byte) (n int, err error) {
 	// Work around this by blocking here and waiting for flushes.
 	// https://github.com/pion/sctp/issues/77
 	c.flushc.L.Lock()
+
+	// Passing func that gets channel ch that signals when
+	// Signal or Broadcast is called on CondChan
 	for c.d.BufferedAmount() > c.d.BufferedAmountLowThreshold() {
-		c.flushc.Wait()
+		c.flushc.Select(func(ch <-chan struct{}) { // Waiting with select
+			select {
+			case <-ch: // Never ending wait
+			case <-c.ctx.Done():
```

Comment:
Why get ctx from c instead of passing it into Write? If you pass it in here you can remove ctx from Wormhole.

Reply:
In this case I was attempting to link the `Context` with its singular `cancel()` function. Since `cancel()` is no longer singular there is no reason to require a singular `Context`, and can allow for `New()` and `Write()` to take separate `Context` parameters.

Reply:
Update: the `Write()` function cannot take a `Context` parameter when used in `io.CopyBuffer()` and other functions assuming the standard `Write` interface. We can optimize the `Wormhole` struct only by breaking general patterns, which I don't recommend.